relay-server/src/services/processor/span/processing.rs

```diff
@@ -253,7 +253,11 @@ pub fn extract_from_event(
         return;
     };
 
-    if state.spans_extracted {
+    let Some(event_headers) = state.event_headers.as_mut() else {
+        return;
+    };
```

Comment:
Should we log an error here if we don't have any headers? This seems very error prone and is going back to the problem we tried to solve (implicit optional state).